vmaas/reposcan/database/csaf_store.py

```diff
@@ -46,7 +48,129 @@ def _save_csaf_files(self, csaf_files: CsafFiles) -> list[int]:
+        for product_id, cpe, package, module in cur.fetchall():
+            if delete:
+                del self.csaf_product_map[(cpe, package, module)]
+            else:
+                self.csaf_product_map[(cpe, package, module)] = product_id
+
+    def _get_cve_rows(self, csaf_cves):
+        # TODO: Known_affected are only parsed as of now. Update here accordingly.
+        product_status_id = 4
+
+        to_insert = []
+        to_update = []
+        for cve, csaf_products in csaf_cves.items():
+            for csaf_product in csaf_products:
+                product_id = self.csaf_product_map.get(csaf_product.to_tuple())
+                if not product_id:
+                    self.logger.error(
+                        "CSAF product: %s associated with CVE: %s not found in the database.", csaf_product.to_tuple(), cve)
+                    continue
+                cve_in_cache = self.csaf_cve_map.get((cve, product_id))
+                if not cve_in_cache:
+                    to_insert.append((cve, product_id, product_status_id))
+                    # Insert key to the cache to avoid duplicates.
+                    # CVE ID will be populated after row is inserted into DB.
+                    self.csaf_cve_map[(cve, product_id)] = (-1, product_status_id)
+                else:
+                    cache_product_status_id = cve_in_cache[1]
+                    if product_status_id != cache_product_status_id:
+                        cve_id = cve_in_cache[0]
+                        to_update.append((cve_id, cve, product_id, product_status_id))
+
+        return to_insert, to_update
+
+    def _cve_refresh_cache(self, cur, delete=False):
+        for cve_id, cve, product_id, product_status_id in cur.fetchall():
+            if delete:
+                del self.csaf_cve_map[(cve, product_id)]
+            else:
+                self.csaf_cve_map[(cve, product_id)] = (cve_id, product_status_id)
+
+    def _populate_cves(self, csaf_cves):
+        self.csaf_cve_map = self._prepare_table_map(cols=["cve", "csaf_product_id"],
+                                                    to_cols=["id", "product_status_id"],
+                                                    table="csaf_cves")
+
+        self._populate_data(csaf_cves, "csaf_cves", ["cve", "csaf_product_id", "product_status_id"],
+                            self._get_cve_rows, self._cve_refresh_cache,
+                            "(%s::int, %s, %s::int, %s::int)")
+
+    def _populate_data(self, csaf_cves, table_name, cols, get_rows, refresh_cache, update_tmpl_str):
+        """Generic method to populate table with CSAF data."""
+        to_insert, to_update = get_rows(csaf_cves)
+        try:
+            cur = self.conn.cursor()
+            if to_insert:
+                execute_values(cur, f"""insert into {table_name} ({', '.join(cols)}) values %s
+                                        returning id, {', '.join(cols)}""",
```

Comment:
returning statement is never used, you are not fetching any values from `execute_values` and not storing it to any variable, please remove `returning ...` if you are not using it

Reply:
it is used in refresh cache

Reply:
sorry I don't see this assigned anywhere, not even fetch param is used
>fetch – if True return the query results into a list (like in a [fetchall()](https://www.psycopg.org/docs/cursor.html#cursor.fetchall)). Useful for queries with RETURNING clause.
 I'm just referring to docs
I'm just curious, how does the result from returning from this query is used in refresh cache, is there some magic which assigns it to the dict?

Reply:
Please try to be more professional in the future. Misunderstanding is a normal thing and can be resolved without being mean or sarcastic. Keep in mind your actions can negatively impact well being of others.

Those parameters are used in this line:
https://github.com/RedHatInsights/vmaas/blob/7f2134f71174ab68747127c0502f0bf1141d11e7/vmaas/reposcan/database/csaf_store.py#L109

Reply:
There was nothing mean, sarcastic or unprofessional in the comment, I'm sorry you understood it this way. I'm seriously asking how rows from returning are being populated, by 'magic' I mean python magic methods which are called on the background, sometimes as a side effect, it's a widely used term. I was asking if there is some magic that execute_values does that it populates dict, that's all.

Thank you for providing me the answer, now I see that you are getting these values using cur.fetchall(). There is a small pitfall with this approach though. cur.fetchall() will fetch only items from the last page. I think it would be safer to do it this way
```
rows = execute_values(cur, "insert into ... returning id, other_rows", fetch=True)
```

Reply:
So you just answered yourself why `page_size=len(to_insert)`. It is done this way in the other parts of the code as well and it has been reviewed and approved by other team members. Maybe there is room for improvement, maybe there isn't. There are more important things to focus on right now, for example fetching product status id in parser, which was your job and was done incompletely. You would be able to fix that during the time you wasted on making those comments. Please prioritize your actions to align better with team objectives.